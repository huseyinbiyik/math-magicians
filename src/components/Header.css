.container {
  width: 90%;
  display: flex;
  justify-content: space-between;
  align-items: baseline;
  margin: 0 auto;
}

.menu {
  display: flex;
  list-style: none;
  gap: 15px;
}

.menu > li {
  padding: 0 10px;
  border-right-width: 1.5px;
  border-right-color: black;
  border-right-style: solid;
}

.menu > li:last-child {
  border: none;
}
