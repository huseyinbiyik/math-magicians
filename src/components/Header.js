import React from 'react';
import { Link } from 'react-router-dom';

export default function Header() {
  return (
    <header className='container'>
      <h1>Math Magician</h1>
      <nav>
        <ul>
          <li>
            <Link to='/home'>Home</Link>
          </li>
          <li>
            <Link to='/calculator'>Calculator</Link>
          </li>
          <li>
            <Link to='/quote'>Quote</Link>
          </li>
        </ul>
      </nav>
    </header>
  );
}
