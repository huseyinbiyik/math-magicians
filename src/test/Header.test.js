import renderer from 'react-test-renderer';
import Header from '../components/Header'
import { HashRouter as Router } from 'react-router-dom';



describe('Test for rendering the Header component', () => {
  it('It should render header ', () => {
    const home = renderer
      .create(<Router><Header /></Router>)
      .toJSON();
    expect(home).toMatchSnapshot();
  });
});
