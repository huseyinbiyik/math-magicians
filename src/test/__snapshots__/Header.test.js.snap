// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for rendering the Header component It should render header  1`] = `
<header
  className="container"
>
  <h1>
    Math Magicians
  </h1>
  <nav>
    <ul
      className="menu"
    >
      <li>
        <a
          href="#/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="#/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          href="#/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
